WCFDataService/Client/System/Data/Services/Client/DataServiceRequest.cs

```diff
@@ -272,6 +273,10 @@ internal long GetQuerySetCount(DataServiceContext context)
 
                 throw;
             }
+            finally
+            {
```

Comment:
Sorry, I don't get that. As far as I can see the line you are referring to 276-277 is within GetQuerySetCount. GetQuerySetCount() is not called from all code paths, Am I on the wrong code line?